testbeam_analysis/tools/plot_utils.py

```diff
@@ -91,19 +92,19 @@ def plot_masked_pixels(input_mask_file, pixel_size=None, dut_name=None, output_p
         ax.set_ylim(-0.5, occupancy.shape[0] - 0.5)
         ax.set_xlabel("Column")
         ax.set_ylabel("Row")
-        output_pdf.savefig()
+        output_pdf.savefig(fig)
 
-        plt.figure()
-        ax = plt.subplot(111)
+        fig = Figure()
+        _ = FigureCanvas(fig)
```

Comment:
Without `FigureCanvas` no `savefig()`. But this seems to work fine in pyBAR without GUI.